sympy/solvers/tests/test_solvers.py

```diff
@@ -1310,6 +1310,10 @@ def test_checksol():
     assert checksol([x - 1, x**2 - 1], x, 1) is True
     assert checksol([x - 1, x**2 - 2], x, 1) is False
     assert checksol(Poly(x**2 - 1), x, 1) is True
+    assert checksol(0, {}) is True
+    assert checksol([1e-10, x - 2], x, 2) is True
```

Comment:
Ideally this should fail.

Reply:
Which assertion are you referring to exactly?

Reply:
The one immediately above with `1e-10` which is not zero.

Reply:
The reason I added that test is because `checksol` seems to treat equations with numeric absolute values less than `1e-9` (after the solution has been substituted in) as zero when checking solutions numerically. On master, something like `checksol(x+1e-10, x, 0)` returns `True`. I added this to test the same behavior for when the equation is just a `float`.

Reply:
I can see the potential merit of doing this when there is a need to substitute something in and numerically evaluate but I think if the input is a plain number then it should just be `return num == 0`.

Reply:
Agreed. The function is checking a solution. Such non-equations would/should have been eliminated before sending here. If there isn't a test like `assert checksol(x+1e-10, x, 0, numerical=True) == True` and `assert checksol(x+1e-10, x, 0, numerical=False) == False` then you could add those.

Reply:
OK. I'll make the change in `checksol` and add the tests you suggest if they don't already exist.

Reply:
I've gone with your suggestion about returning `f.is_zero` if `f.free_symbols` is empty, and I've added the tests you suggested, so I'm resolving this.